### PR TITLE
Get redgreen-optimized working on Cray/AMD

### PR DESCRIPTION
**Pull request by samhatfield**
redgreen-optimized works great on NV-platforms. Cray/AMD platforms like LUMI are still showing issues. This PR will try to collect all changes needed to get this branch working on these platforms without compromising support for NV.

### CODE REVIEW

**Comment by samhatfield** [comment]
Current situation
-----------------

| Commit | AC (Nvidia) Status | LUMI (Cray/AMD)
| ------ | --------|------
| 1adddfc (last commit before PR) | Works | Does not compile -> type incompatibility in `HICBLAS_MOD` (`ACC_GET_HIP_STREAM`)
| 27644e4 | Works (bit identical with previous) | Compiles but segfaults when running -> GEMM graph implementation suspected
| 9bb1ac4 | Works but gives NaNs | `HIPFFT_PARSE_ERROR`

Probably something wrong with `9bb1ac47643aa3877c83ffec04549b0327364f87` (graph disabling). Need to get this working on AC before continuing.

**Comment by samhatfield** [comment]
| Commit | AC (Nvidia) Status | LUMI (Cray/AMD)
| ------ | --------|------
| bddff0b | Works (bit identical with 27644e4) | `HIPFFT_PARSE_ERROR`

Progress...

File: src/trans/gpu/algor/module/hicblas_mod.F90 [hunk 1]
```diff
@@ -258,9 +258,13 @@ SUBROUTINE HIP_DGEMM_BATCHED_OVERLOAD( &
...
     INTEGER(KIND=JPIM) :: STRIDEC
     INTEGER(KIND=JPIM) :: BATCHCOUNT
-    INTEGER(KIND=C_LONG) :: STREAM
+    INTEGER(KIND=C_INT) :: STREAM
     TYPE(GROWING_ALLOCATION_TYPE), INTENT(IN) :: ALLOC
 
+    INTEGER(KIND=C_LONG) :: HIP_STREAM
+
+    HIP_STREAM = INT(ACC_GET_HIP_STREAM(STREAM), C_LONG)
+
```
Comment (marsdeno) [inline src/trans/gpu/algor/module/hicblas_mod.F90:246]: Neat.

**Comment by samhatfield** [comment]
Okay, @marsdeno assuming you're happy with this I think you might as well merge it so I can get on to Paul.

**Comment by marsdeno** [comment]
OK thanks a lot Sam

**Pull request merged**